controllers/rhmi/rhmi_controller.go

```diff
@@ -224,6 +229,8 @@ func New(mgr ctrl.Manager) *RHMIReconciler {
 
 // +kubebuilder:rbac:groups=operator.marin3r.3scale.net,resources=discoveryservices,verbs=get;list;watch;create;update;delete,namespace=integreatly-operator
 
+// +kubebuilder:rbac:groups=operator.openshift.io,resources=cloudcredentials,verbs=get;list;watch
```

Comment:
Considering that there will be just one CloudCredentials CR on the cluster, we can leave this code as is.
So just for future reference: we can limit the permissions to just `get` on a CR with specific name("cluster"):
```
// +kubebuilder:rbac:groups=operator.openshift.io,resources=cloudcredentials,verbs=get,resourceNames=cluster
```
That will work if you create a new client that doesn't use cache, e.g.:
```
serverClient, err := k8sclient.New(r.restConfig, k8sclient.Options{
  Scheme: r.mgr.GetScheme(),
})
// handle err
```
and pass that^ client to `checkIfStsClusterByCredentialsMode`

Reply:
Thank you very much for comment. Will use for future.